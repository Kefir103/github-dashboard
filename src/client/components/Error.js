import React from 'react';
import { connect } from 'react-redux';

function Error(props) {
    return (
        <div className={'app-container error'}>
            <h1>Упс :(</h1>
            <h2>Произошла ошибочка...</h2>
            {props.catchedError ? (
                <h3>
                    ({props.catchedError.status} ({props.catchedError.statusText}))
                </h3>
            ) : (
                ''
            )}
        </div>
    );
}

const mapStateToProps = (state) => {
    return {
        catchedError: state.appStatus.catchedError,
    };
};

export default connect(mapStateToProps)(Error);
