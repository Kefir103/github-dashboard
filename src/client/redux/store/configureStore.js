import { createStore, applyMiddleware, compose } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import { rootReducer } from '../reducers/rootReducer';

let store;

export default function configureStore(initialState) {
    if (store) {
        return store;
    }

    store =
        process.env.NODE_ENV === 'production'
            ? createStore(rootReducer, initialState, compose(applyMiddleware(thunk)))
            : createStore(rootReducer, initialState, composeWithDevTools(applyMiddleware(thunk)));

    return store;
}
