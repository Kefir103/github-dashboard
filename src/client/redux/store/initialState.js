export const initialState = {
    appStatus: {
        isLoading: true,
    },
    repository: {
        repos: [],
        totalCount: 0,
        contributors: [],
        currentRepository: {},
    },
    filter: {
        searchText: '',
        currentPage: 1,
    },
};
