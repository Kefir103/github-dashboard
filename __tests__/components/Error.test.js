import React from 'react';
import Error from '../../src/client/components/Error';
import { initialState } from '../../src/client/redux/store/initialState';
import configureMockStore from 'redux-mock-store';
import { shallow, render } from 'enzyme';

const mockStore = configureMockStore();

describe('<Error />', () => {
    let store;
    beforeEach(() => {
        store = mockStore({ ...initialState });
    });
    test('should be rendered correctly with catchedError', () => {
        const mockingState = {
            appStatus: {
                catchedError: {
                    status: 403,
                    statusText: 'statusText',
                },
            },
        };
        store = mockStore({
            ...initialState,
            ...mockingState,
        });
        const wrapper = shallow(<Error store={store} />);
        const component = render(wrapper);
        expect(component).toMatchSnapshot();
    });
    test('should be rendered correctly without catchedError', () => {
        const wrapper = shallow(<Error store={store} />);
        const component = render(wrapper);
        expect(component).toMatchSnapshot();
    });
});
