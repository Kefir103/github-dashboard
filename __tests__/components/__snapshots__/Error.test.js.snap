// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should be rendered correctly with catchedError 1`] = `
<div
  class="app-container error"
>
  <h1>
    Упс :(
  </h1>
  <h2>
    Произошла ошибочка...
  </h2>
  <h3>
    (403 (statusText))
  </h3>
</div>
`;

exports[`<Error /> should be rendered correctly without catchedError 1`] = `
<div
  class="app-container error"
>
  <h1>
    Упс :(
  </h1>
  <h2>
    Произошла ошибочка...
  </h2>
</div>
`;
